src/main/java/Asta/Asta.java

```diff
@@ -0,0 +1,116 @@
+package Asta;
+
+import Asta.command.Command;
+import Asta.command.Parser;
+import Asta.task.Task;
+import Asta.task.TaskList;
+
+import java.util.ArrayList;
+
+public class Asta {
+    private static final String FILE_PATH = "./data/asta.txt";
+    private final Storage storage;
+    private final Ui ui;
+    private TaskList tasks;
+
+    public Asta(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (AstaException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * The entry point of the application.
+     *
+     * @param args Command line arguments (not used).
+     */
+    public static void main(String[] args) {
+        new Asta(FILE_PATH).run();
+    }
+
+    /**
+     * Runs the main logic of the application. Reads user input and processes commands.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            String fullCommand = ui.readCommand();
+            Command command = Parser.parse(fullCommand);
+            ui.showLine();
+            try {
+                switch (command) {
```

Comment:
Nice switch statement! I see that there is a lot of handler logic nested in each case, do you think that it might be a good idea to create dedicated handler methods for each of the commands?